frontend/src/App.test.tsx

```diff
@@ -327,6 +399,29 @@ describe("App", () => {
     expect(wrapper.find(Modal)).toHaveLength(1)
   })
 
+  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when the app is first rendered", () => {
+    const props = getProps()
+    shallow(<App {...props} />)
+
+    expect(props.hostCommunication.sendMessage).toHaveBeenCalledWith({
+      type: "SCRIPT_RUN_STATE_CHANGED",
+      scriptRunState: ScriptRunState.NOT_RUNNING,
+    })
+  })
+
+  it("sends SCRIPT_RUN_STATE_CHANGED signal to the host when scriptRunState changing", () => {
+    const props = getProps()
+    const wrapper = shallow(<App {...props} />)
+
+    Object.values(ScriptRunState).forEach(scriptRunState => {
+      wrapper.setState({ scriptRunState })
+      expect(props.hostCommunication.sendMessage).toHaveBeenCalledWith({
+        type: "SCRIPT_RUN_STATE_CHANGED",
+        scriptRunState,
+      })
+    })
+  })
+
```

Comment:
Would be good to have another test that the message isn't sent when the script run state doesn't change